### PR TITLE
AutoSubscribe subscribes concurrently

### PR DESCRIPTION
**Pull request by danielmarbach**
Closes https://github.com/Particular/NServiceBus/issues/4897

Targets master, can be retargeted to patch or minor (would prefer patch to get the user unstuck). Does not require backporting.

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/nservicebus-maintainers OK if I target this against 6.3.5 to get things moving?

**Comment by timbussmann** [comment]
@danielmarbach please do not target 6.3.5 directly yet as we have a couple of patches to be released very soon.

**Comment by bording** [comment]
LGTM, though we'll need to decide how to ship it.

File: src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs [hunk 1]
```diff
@@ -59,26 +59,33 @@ static List<Type> GetMessageTypesHandledByThisEndpoint(MessageHandlerRegistry ha
...
             }
 
-            IEnumerable<Type> messagesHandledByThisEndpoint;
+            static async Task SubscribeToEvent(IMessageSession session, Type eventType)
+            {
+                await session.Subscribe(eventType).ConfigureAwait(false);
+                Logger.DebugFormat("Auto subscribed to event {0}", eventType);
+            }
+
+            List<Type> messagesHandledByThisEndpoint;
```
Comment (timbussmann) [inline src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs]: would an array be more efficient?

**Comment by timbussmann** [comment]
I retargeted the PR to target the 6.4 release branch

File: src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs (hunk 1 above)
Comment (bording) [inline src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs]: There's a bit less overhead in an array compared to a list, but this also isn't a per-message execution path, so I don't think it's worth changing.

**Comment by bording** [comment]
FYI, when this is merged into the `release-6.4` branch, be sure to do a squash and merge on it!

**Pull request merged**